src/detect-http-header-names.c

```diff
@@ -410,6 +613,16 @@ void DetectHttpHeaderNamesRegister(void)
     DetectAppLayerInspectEngineRegister2(BUFFER_NAME, ALPROTO_HTTP1, SIG_FLAG_TOCLIENT,
             HTP_RESPONSE_HEADERS, InspectEngineHttpHeaderNames, NULL);
 
+    DetectAppLayerMpmRegister2(BUFFER_NAME, SIG_FLAG_TOSERVER, 2,
```

Comment:
seems that you should be able to use the generic implementations for prefilter and inspect here. I don't see any specific reason for a custom implementation. Or am I missing something?

Reply:
I think you are right, this was copy pasted from the `DetectAppLayerMpmRegister2` HTTP1 line above...